include/grpc/grpc_security.h

```diff
@@ -775,6 +775,15 @@ typedef struct grpc_tls_certificate_provider grpc_tls_certificate_provider;
  */
 typedef struct grpc_tls_identity_pairs grpc_tls_identity_pairs;
 
+/**
+ * Status generated for certificate-key match.
+ */
+typedef struct grpc_tls_status_or_bool {
```

Comment:
I initially used this approach, but I ran into some issues. I approached @ZhenLian to fix it, and we decided to go for the approach I am currently using. The main reason was that we didn't want to put more work/stress/effort on the user as they would need to be very careful with these pointers. 
Although this struct might not be that useful for other APIs, it suits this one fine and we decided to go with that

Reply:
Hmm, I don't see much benefit because instead of freeing error details pointer, user has to make sure to invoke the release API. 
Similar structure is also used in in-memory PR, resulting in code replication. 
I would still push for my recommendation, but if current structure is what we decide to use, I think we should at-least make it a general structure, used by both key-match and in-memory (and more), and even the release API should have a general naming.
@ZhenLian Wdyt?

Reply:
Yeah, we can have a general structure in the core API - that's actually what I've planned for. Whichever PR goes in first can add the struct and the latter can just reuse it. No need to create a separate struct for that.
The reason why I didn't choose to put `code` and `error_details` as input parameter but instead making them a return value is that, considering how users would use this API:
```
grpc_status_code status;
char* error_details = nullptr; // Not entirely sure if it works. maybe we need gpr_strdup() here to create a dummy string?
bool result = grpc_tls_certificate_key_match(..., &status, &error_details);
if (error_details != nullptr) {
  gpr_free(error_details);  ​
}
```
I think the condition doesn't get better, because callers still need to remember to free the error message at the end. It actually makes it a bit worse we will ask callers to explicitly free it  using `gpr_free`, not something else, because we will need to use `gpr_strdup` inside. 
This is an interesting problem, and I agree this applies to in-memory PR as well. Let me ask the gRPC team for advice and update here.

Reply:
OK I got replies from gRPC team and they actually prefer something like this:
```
GRPCAPI grpc_status_code grpc_tls_certificate_key_match(..., char** error_details)
```
We will need to make sure in the document very clear that the caller will need to take ownership of the error message string and free it via gpr_free().
@itsemmanuelfrancis we will need similar changes for in-memory provider as well.